privilege: add REPAIRCLUSTERMETADATA system prvilege and remove admin checks

rafiss opened the pull request:
As we move away from requiring the admin role to perform cluster debug/repair operations, we want to use a privilege instead. To facilitate that, the admin role now implicitly has ALL global privileges. The privilege for admins is not revokeable.

Some exceptions were made, since the focus of this change is to use this
privilege for debug/repair endpoints. The following are still gated by
the admin role:
- ALTER ROLE on an admin
- ALTER DEFAULT PRIVILEGES FOR ALL ROLES
- changing a backup schedule metric

informs https://github.com/cockroachdb/cockroach/issues/109814

Release note (sql change): SQL debug/repair commands that were previously only usable
by users with the admin role can now be used by users with the
VIEWCLUSTERMETADATA or REPAIRCLUSTERMETADATA system privilege, depending
on whether the operation is read-only or modifies state.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110084)
<!-- Reviewable:end -->

rafiss committed:
privilege: add REPAIRCLUSTERMETADATA system prvilege

Release note: None

rafiss committed:
*: eliminate usages of planner.RequireAdminRole

All the usages are replaced by checks for VIEWCLUSTERMETADATA or
REPAIRCLUSTERMETADATA. admins implicitly have this privilege, so this
should not break any existing usage.

Some exceptions were made, since the focus of this change is to use this
privilege for debug/repair endpoints. The following are still gated by
the admin role:
- ALTER ROLE on an admin
- ALTER DEFAULT PRIVILEGES FOR ALL ROLES
- changing a backup schedule metric

Release note (sql change): SQL commands that were previously only usable
by users with the admin role can now be used by users with the
VIEWCLUSTERMETADATA or REPAIRCLUSTERMETADATA system privilege, depending
on whether the operation is read-only or modifies state.

rafiss commented:
tftr!

bors r+
